src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs

```diff
@@ -210,17 +210,18 @@ public class Task : IAsyncResult, IDisposable
 
         // This dictonary relates the task id, from an operation id located in the Async Causality log to the actual
         // task. This is to be used by the debugger ONLY. Task in this dictionary represent current active tasks.
-        private static readonly Dictionary<int, Task> s_currentActiveTasks = new Dictionary<int, Task>();
-        private static readonly object s_activeTasksLock = new object();
+        private static Dictionary<int, Task> s_currentActiveTasks;
 
         // These methods are a way to access the dictionary both from this class and for other classes that also
         // activate dummy tasks. Specifically the AsyncTaskMethodBuilder and AsyncTaskMethodBuilder<>
         internal static bool AddToActiveTasks(Task task)
         {
             Debug.Assert(task != null, "Null Task objects can't be added to the ActiveTasks collection");
 
+            LazyInitializer.EnsureInitialized(ref s_currentActiveTasks, () => new Dictionary<int, Task>());
```

Comment:
The delegate passed as the second argument isn't necessary.  It can just be:
```C#
LazyInitializer.EnsureInitialized(ref s_currentActiveTasks);
```

Reply:
I am not a fan of that as that brings unnecessary dynamic lookup, instead of simple static checks done by compiler

Reply:
Ok.  The downside to this, in contrast, is that every call needs to read and null check the static, e.g. it basically compiles to:
```C#
LazyInitializer.EnsureInitialized(ref s_currentActiveTasks, s_cachedDelegate ?? (s_cachedDelegate = LambdaMethod));
```
whereas that doesn't happen with the overload that just takes the ref.  However, in this case I don't mind because this AddToActiveTasks is already a slower path that's only used when a debugger is attached.